### ISSUE TITLE
How do we get the specific value of the dimension of y_true when we customize the loss function?

### ISSUE THREAD

**Issue by yansudong** [keras-team/keras#12680]
When I customize the loss function, I want to get the specific value of the y_true dimension. 
I tried y_true.shape, y_true.get_shape(), y_true.get_shape().as_list(), and K.shape(y_true), but Will report an error because of data format and other issues. 
Does anyone have relevant experience? Who can help me? thank you very much!
![image](https://user-images.githubusercontent.com/31694573/56200675-cc7fa880-6071-11e9-80db-d9a0b9de1910.png)

**Comment by yansudong** [comment]
This is my custom loss function:
`def my_newloss(y_true, y_pred, alpha=0.8):
    mseloss = K.mean(K.square(y_pred - y_true), axis=-1)
    num = (K.int_shape(y_true)[0]) * (K.int_shape(y_true)[1])
    samm = K.zeros(num)
    n = 0
    for x in range(K.int_shape(y_true)[0]):
        for y in range(K.int_shape(y_true)[1]):
            z = K.reshape(y_pred[ x, y,:], [-1])
            sa = K.reshape(y_true[x, y,:], [-1])
            tem1=K.dot(z,sa)
            tem2=(K.linalg.norm(z))*(K.linalg.norm(sa))
            samm[n]=K.acos(tem1/tem2)
            n=n+1
    SAM=(K.mean(samm))*180/math.pi
    samloss = SAM
    return alpha*mseloss + (1-alpha)*samloss`
Who can help me solve it? The main problem is the calculation and coding of num and samm.

**Comment by yansudong** [comment]
This is my custom loss function:

```Python
def my_newloss(y_true, y_pred, alpha=0.8):
    mseloss = K.mean(K.square(y_pred - y_true), axis=-1)
    num = (K.int_shape(y_true)[0]) * (K.int_shape(y_true)[1])
    samm = K.zeros(num)
    n = 0
    for x in range(K.int_shape(y_true)[0]):
        for y in range(K.int_shape(y_true)[1]):
            z = K.reshape(y_pred[ x, y,:], [-1])
            sa = K.reshape(y_true[x, y,:], [-1])
            tem1=K.dot(z,sa)
            tem2=(K.linalg.norm(z))*(K.linalg.norm(sa))
            samm[n]=K.acos(tem1/tem2)
            n=n+1
    SAM=(K.mean(samm))*180/math.pi
    samloss = SAM
    return alpha*mseloss + (1-alpha)*samloss

```
Who can help me solve it? The main problem is the calculation and coding of num and samm.

**Comment by ymodak** [comment]
Hi!
This issue isn't related to a bug/enhancement/feature request or other accepted types of issue.
To ask questions, please see the following resources :
[StackOverflow](http://stackoverflow.com/questions/tagged/keras)
[Keras Slack channel](https://keras-slack-autojoin.herokuapp.com)
[Keras.io support](https://keras.io/#support)
[Gitter](https://gitter.im/Keras-io/Lobby)
Thanks!
If you think we made a mistake, please open another issue explaining your request in detail.
